src/rules/expect.ts

```diff
@@ -123,15 +117,7 @@ function validate(context: TSESLint.RuleContext<MessageIds, [Options]>, options:
     });
   }
 
-  const seenDiagnosticsOnLine = new Set<number>();
-
-  for (const diagnostic of diagnostics) {
-    const line = lineOfPosition(diagnostic.start!, sourceFile);
-    seenDiagnosticsOnLine.add(line);
-    if (!errorLines.has(line)) {
-      addDiagnosticFailure(diagnostic);
-    }
-  }
+  const seenDiagnosticsOnLine = new Set(diagnostics.map((diagnostic) => lineOfPosition(diagnostic.start!, sourceFile)));
```

Comment:
Do you know why `diagnostic.start` is nullable / optional? Add a comment about why this is safe.

Reply:
I honestly have no idea.